api/src/test/java/org/openmrs/module/idgen/SequentialIdentifierGeneratorTest.java

```diff
@@ -73,4 +94,49 @@ public void shouldSetNextSequenceValueToNegative() throws Exception {
 		generator.setMaxLength(13);
  		assertThat(generator.getNextSequenceValue(), is(-1l));
 	}
+	
+	@Test
+	public void getIdentifierForSeed_shouldGenerateLocationPrefixedIdFromLocationBasedPrefixProvider() {
+		SequentialIdentifierGenerator generator = new SequentialIdentifierGenerator();
+		generator.setBaseCharacterSet("0123456789");
+		generator.setFirstIdentifierBase("000");
+		generator.setName("Location Prefixed Sequential Identifier Source");
+		generator.setPrefixProviderBean(LocationBasedPrefixProvider.class.getSimpleName());
+		
+		UserContext userContext = mock(UserContext.class);
+		when(userContext.getLocation()).thenReturn(createLocationTree());
+		when(Context.getUserContext()).thenReturn(userContext);
+		when(Context.getRegisteredComponent("LocationBasedPrefixProvider", PrefixProvider.class)).thenReturn(new LocationBasedPrefixProvider());
+		
+		assertThat(generator.getIdentifierForSeed(1L), is("AFDEL-001"));
+	}
+	
+	private Location createLocationTree() {
+		LocationAttributeType prefixAttrType = new LocationAttributeType();
+		prefixAttrType.setName(IdgenConstants.PREFIX_LOCATION_ATTRIBUTE_TYPE);
+		prefixAttrType.setMinOccurs(0);
+		prefixAttrType.setMaxOccurs(1);
+		prefixAttrType.setDatatypeClassname("org.openmrs.customdatatype.datatype.FreeTextDatatype");
```

Comment:
As per my suggestion about the activator, you would not this one anymore and could just do somewhere in your setup:
```java
(new IdgenModuleActivator()).initPrefixAttributeType();
```
If that doesn't work here you can make a bean of your activator, like [here](https://github.com/openmrs/openmrs-module-attachments/blob/1ab881ac7d6b2aadb08276e27a6fcf7cc7923343/api/src/main/java/org/openmrs/module/attachments/AttachmentsActivator.java#L33-L34), and then Spring-wire it.

Reply:
Can we leave this as is? This method is a DB transactional method and I failed extending the `BaseModuleContextSensitiveTest` when this testcase is run by PowerMockRunner (`@RunWith(PowerMockRunner.class)`) Is it something we can't ignore?

Reply:
Ok so this is not a Spring Test, so you can just do the same as suggested above for `LocationBasedPrefixProviderTest`:
```java
(new IdgenModuleActivator()).initPrefixAttributeType();
```